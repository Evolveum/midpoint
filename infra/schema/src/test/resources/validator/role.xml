<!-- invalid OID format -->
<role xmlns="http://midpoint.evolveum.com/xml/ns/public/common/common-3"
        xmlns:t="http://prism.evolveum.com/xml/ns/public/types-3"
        xmlns:q="http://prism.evolveum.com/xml/ns/public/query-3"
        oid="6d1a2464-db78-43e8-b1f1-cdf73cd54fxx">

    <name>validation role</name>

    <!-- deprecated item -->
    <subtype>deprecated item</subtype>

    <parentOrgRef oid="00000000-8888-6666-0000-100000000001" type="OrgType"/>
    <!-- multivalue ref without oid -->
    <parentOrgRef>
        <filter>
            <q:text>name = "sample"</q:text>
        </filter>
    </parentOrgRef>

    <credentials>
        <password>
            <value>
                <!-- clear value, external data shoud be there -->
                <t:clearValue>qwe123</t:clearValue>
            </value>
        </password>
    </credentials>

    <!-- missing natural key -->
    <inducement>
        <!-- invalid OID format -->
        <targetRef oid="73ca6aa2-e8ac-4097-a9b7-ca539dec45xx" type="RoleType"/>
    </inducement>
</role>
