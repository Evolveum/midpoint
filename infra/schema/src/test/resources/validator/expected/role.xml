<!--
  ~ Copyright (C) 2010-2023 Evolveum and contributors
  ~
  ~ This work is dual-licensed under the Apache License 2.0
  ~ and European Union Public License. See LICENSE file for details.
  -->

<role xmlns="http://midpoint.evolveum.com/xml/ns/public/common/common-3"
        oid="f4244705-69a7-4602-95ad-6a4ad7beddb7">

    <name>role</name>

    <subtype>some deprecated type</subtype>

    <assignment id="1">
        <personaConstruction>
            <targetType>some-type</targetType>
            <targetSubtype>some-subtype</targetSubtype>
        </personaConstruction>
    </assignment>

    <assignment id="2">
        <limitOtherPrivileges>
            <approvalWorkItems>
                <all>true</all>
            </approvalWorkItems>
        </limitOtherPrivileges>
    </assignment>

    <inducement id="3">
        <limitOtherPrivileges>
            <approvalWorkItems>
                <all>true</all>
            </approvalWorkItems>
        </limitOtherPrivileges>
    </inducement>

    <inducement id="4">
        <limitOtherPrivileges>
            <approvalWorkItems>
                <all>true</all>
            </approvalWorkItems>
            <caseManagementWorkItems>
                <all>false</all>
            </caseManagementWorkItems>
        </limitOtherPrivileges>
    </inducement>
</role>
